akka-discovery/src/main/scala/akka/discovery/ServiceDiscovery.scala

```diff
@@ -119,6 +119,45 @@ case object Lookup {
    * and protocol
    */
   def create(serviceName: String): Lookup = new Lookup(serviceName, None, None)
+
+  private val SrvQuery = """^_(.+?)\._(.+?)\.(.+?)$""".r
+
+  private val DomainName = "^((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{2,6}$".r
+
+  /**
+   * Create a service Lookup from a string with format:
+   * _portName._protocol.serviceName.
+   * (as specified by https://www.ietf.org/rfc/rfc2782.txt)
+   *
+   * If the passed string conforms with this format, a SRV Lookup is returned.
+   * The serviceName part must be a valid domain name.
+   *
+   * The string is parsed and dismembered to build a Lookup as following:
+   * Lookup(serviceName).withPortName(portName).withProtocol(protocol)
+   *
+   * If the string doesn't not conform with the SRV format, a simple A/AAAA Lookup is returned
+   * using the whole string as service name.
+   */
+  def fromString(str: String): Lookup =
+    str match {
+      case SrvQuery(portName, protocol, serviceName) if validDomainName(serviceName) ⇒
+        Lookup(serviceName).withPortName(portName).withProtocol(protocol)
+
+      case _ ⇒ Lookup(str)
```

Comment:
Here we can be potentially returning an invalid Lookup? I have not yet looked at that `apply`.

Reply:
yes, there is no check on the lookup. If one passes an invalid name they will find nothing. 

I'm not so sure if we want to be strick on this. What is an invalid name? We know what is an invalid one in the context of SRV because there is a spec for it, but maybe we should not be strict in the impl of `Lookup.apply`

Reply:
I wonder if it should be Option[Lookup]? or add InvalidLookup? @raboof or @chbatey thoughts?